searches/fibonacci_search.py

```diff
@@ -1,51 +1,130 @@
+
+    >>> fibonacci(0)
+    0
+    >>> fibonacci(2)
+    1
+    >>> fibonacci(5)
+    5
+    >>> fibonacci(15)
+    610
+    >>> fibonacci('a')
+    Traceback (most recent call last):
+    TypeError: k must be an integer.
+    >>> fibonacci(-5)
+    Traceback (most recent call last):
+    ValueError: k integer must be greater or equal to zero.
     """
-    >>> fibonacci_search([1,6,7,0,0,0], 6)
+    if not isinstance(k, int):
+        raise TypeError("k must be an integer.")
+    if k < 0:
+        raise ValueError("k integer must be greater or equal to zero.")
+    if k == 0:
+        return 0
+    elif k == 1:
+        return 1
+    else:
+        return fibonacci(k - 1) + fibonacci(k - 2)
+
+
+def fibonacci_search(arr: list, val: int) -> int:
+    """A pure Python implementation of a fibonacci search algorithm.
+
+    Parameters
+    ----------
+    arr
+        List of sorted elements.
+    val
+        Element to search in list.
+
+    Returns
+    -------
+    int
+        The index of the element in the array.
+        -1 if the element is not found.
+
+    >>> fibonacci_search([4, 5, 6, 7], 4)
+    0
+    >>> fibonacci_search([4, 5, 6, 7], -10)
+    -1
+    >>> fibonacci_search([-18, 2], -18)
+    0
+    >>> fibonacci_search([5], 5)
+    0
+    >>> fibonacci_search(['a', 'c', 'd'], 'c')
     1
-    >>> fibonacci_search([1,-1, 5, 2, 9], 10)
+    >>> fibonacci_search(['a', 'c', 'd'], 'f')
+    -1
+    >>> fibonacci_search([], 1)
     -1
+    >>> fibonacci_search([.1, .4 , 7], .4)
+    1
     >>> fibonacci_search([], 9)
-    0
+    -1
+    >>> fibonacci_search(list(range(100)), 63)
+    63
+    >>> fibonacci_search(list(range(100)), 99)
+    99
+    >>> fibonacci_search(list(range(-100, 100, 3)), -97)
+    1
+    >>> fibonacci_search(list(range(-100, 100, 3)), 0)
+    -1
+    >>> fibonacci_search(list(range(-100, 100, 5)), 0)
+    20
+    >>> fibonacci_search(list(range(-100, 100, 5)), 95)
+    39
     """
-    fib_N_2 = 0
-    fib_N_1 = 1
-    fibNext = fib_N_1 + fib_N_2
-    length = len(arr)
-    if length == 0:
-        return 0
-    while fibNext < len(arr):
-        fib_N_2 = fib_N_1
-        fib_N_1 = fibNext
-        fibNext = fib_N_1 + fib_N_2
-    index = -1
-    while fibNext > 1:
-        i = min(index + fib_N_2, (length - 1))
-        if arr[i] < val:
-            fibNext = fib_N_1
-            fib_N_1 = fib_N_2
-            fib_N_2 = fibNext - fib_N_1
-            index = i
-        elif arr[i] > val:
-            fibNext = fib_N_2
-            fib_N_1 = fib_N_1 - fib_N_2
-            fib_N_2 = fibNext - fib_N_1
-        else:
-            return i
-    if (fib_N_1 and index < length - 1) and (arr[index + 1] == val):
-        return index + 1
-    return -1
+    len_list = len(arr)
+    # Find m such that F_m >= n where F_i is the i_th fibonacci number.
+    i = 0
+    while True:
+        if fibonacci(i) >= len_list:
+            greater_fibb_index = i
+            break
+        i += 1
+    fibb_k = greater_fibb_index
```

Comment:
```suggestion
            fibb_k = i
            break
        i += 1
```
You can directly assign it in the loop.